### ISSUE TITLE
Upgrade to the torch 1.3

### ISSUE THREAD

**Issue by snarb** [cardwing/Codes-for-Lane-Detection#167]
I have troubles with PyTorch 0.3 and cuda 8 on 2080RTX because of some cuda bug. I got Out of memory during coping model on GPU.  SO I have did some naive upgrade of the code to the PyTorch 1.3.

 I have removed no_partialbn code branch that is using deprecated torch.utils.ffi. It is hard to update it. What would you say about partial batch normalization branch and it necessity?

I am also not using trained model and use smaller learning rate, because of unstable learning and nan loss. 
 Other changes minor. Looks like everything works well. I can create pull request to new branch if this makes sense.

**Comment by cardwing** [comment]
Synchronization BN is useful to the stability of training. I guess the nan loss is caused by freezing BN (i.e., partial BN). You can have a try of pytorch' syn BN. Besides, can you provide the performance of your trained model so that we can know better about the upgraded code? Thanks. ^_^

**Comment by guhaohitsz** [comment]
> I have troubles with PyTorch 0.3 and cuda 8 on 2080RTX because of some cuda bug. I got Out of memory during coping model on GPU. SO I have did some naive upgrade of the code to the PyTorch 1.3.
> 
> I have removed no_partialbn code branch that is using deprecated torch.utils.ffi. It is hard to update it. What would you say about partial batch normalization branch and it necessity?
> 
> I am also not using trained model and use smaller learning rate, because of unstable learning and nan loss.
> Other changes minor. Looks like everything works well. I can create pull request to new branch if this makes sense.

Hi,
     I'm facing the similar situation as you, but I'm new to pytorch and I don't know how to upgrade the code;
     Could you help to send me your changes(guhaohitsz@gmail.com), thanks!

**Comment by guhaohitsz** [comment]
@snarb

**Comment by snarb** [comment]
@guhaohitsz  I will send you. I have also rewrote matlab code in the  python,  added realtime demo on the video, and moved to constants hardcoded numbers like image dimentions .  Also I have added code to fix the case when in one line the can be points from another line. Matlab code doesn't handle it.
@cardwing  Now I am training on my own dataset, I will try to train upgraded model on CULane.

**Comment by TommyAnqi** [comment]
@snarb Hi, would you like to share your upgrade code, I am also processing the raw out from the net, can you give me some detailed advices on the postprocessing module, thanks a lot!

**Comment by TommyTang930** [comment]
@snarb
I hope you can share your upgraded code here. That's really helpful.

**Comment by noobliang** [comment]
> @guhaohitsz I will send you. I have also rewrote matlab code in the python, added realtime demo on the video, and moved to constants hardcoded numbers like image dimentions . Also I have added code to fix the case when in one line in some cases there can be point from another line and result line become wrong. Matlab code doesn't handle it. I can share this changes as well.
> @cardwing Now I am training on my own dataset, I will try to train upgraded model on CULane.

Hi,  have you update the SAD code or loss funtion in ERFNet?  I cant understand the lua code in ENet-label-torch.

**Comment by noobliang** [comment]
@snarb Hi,  I have new to pytorch to 1.1, could you help to share  me you changed code ,thanks a lots .  my email: (lm992496359@gmail.com)

**Comment by sieuwe1** [comment]
@snarb Can you also share your upgraded code with me? Email: sieuwe_crossen@hotmail.com.

**Comment by ixtiyoruz** [comment]
> @guhaohitsz I will send you. I have also rewrote matlab code in the python, added realtime demo on the video, and moved to constants hardcoded numbers like image dimentions . Also I have added code to fix the case when in one line in some cases there can be point from another line and result line become wrong. Matlab code doesn't handle it. I can share this changes as well.
> @cardwing Now I am training on my own dataset, I will try to train upgraded model on CULane.

Hi, Please share your upgraded code with me as well. email:ixtiyoruz312@gmail.com

**Comment by ixtiyoruz** [comment]
if there anybody who is trying to convert it into tensorflow via python. I have implemented it as i could, please refer to [https://github.com/ixtiyoruz/Enet-SAD-tensorflow.git](url) for more details. I haven't get it work as the paper if you find any error please tell me.

**Comment by snarb** [comment]
Hi all I got a lot of requests. I will create PL soon.

**Comment by snarb** [comment]
https://github.com/cardwing/Codes-for-Lane-Detection/pull/186
@guhaohitsz   I have achieved 0.60 mIoU on val CUlane dataset when training 14h+ from scratch on one 2080RTX GPU. It is still slow improving, so I think it should be the same 0.61 as restrained model.

**Comment by cardwing** [comment]
@snarb Thanks for your contribution. However, I wonder what is the performance (i.e., F1-measure) of this erfnet model on CULane testing set.

**Comment by Leviou** [comment]
@snarb  Can I get  your code?  Email: zhaoluswc@gmail.com  Thank you

**Comment by ytusdc** [comment]
@snarb Hi，can you share your upgraded code with me as well. I use pytorch 1.4 and RTX2080 Ti
 email: ytusdc@gmail.com.

**Comment by Ace-Xu** [comment]
@snarb  Thanks for your work, can you also share your code with me?  Email: lxzxxzj@gmail.com.

**Comment by miao02830** [comment]
> > @guhaohitsz我会发给你的。我还在 python 中重写了 matlab 代码，在视频中添加了实时演示，并移至常量硬编码数字，如图像尺寸。我还添加了代码来修复在某些情况下一行中可能有另一行的点并且结果行出错的情况。Matlab 代码不处理它。我也可以分享这些变化。
> > @cardwing现在我正在我自己的数据集上进行训练，我将尝试在 CULane 上训练升级后的模型。
> 
> 您好，您是否更新了 ERFNet 中的 SAD 代码或损失函数？我无法理解 ENet-label-torch 中的 lua 代码。

您好，请问您能分享一下erfnet中的sad代码？谢谢！
